sphinx/builders/html/__init__.py

```diff
@@ -405,6 +407,23 @@ def get_outdated_docs(self) -> Iterator[str]:
 
         if self.templates:
             template_mtime = int(self.templates.newest_template_mtime() * 10**6)
+            try:
+                old_mtime = _last_modified_time(build_info_fname)
+            except Exception:
+                pass
+            else:
+                # Let users know they have a newer template
+                if template_mtime > old_mtime:
+                    logger.info(
+                        bold("building [html]: ") +
+                        __(
+                            "template %s is newer (%s) than previous build info (%s), "
+                            "all docs will be out of date"
+                        ),
+                        self.templates.newest_template_name(),
+                        _format_rfc3339_microseconds(template_mtime),
+                        _format_rfc3339_microseconds(old_mtime),
```

Comment:
Perhaps a more direct message? Though it looses the mtimes.
```suggestion
                        __(
                            "template %s has been changed since the previous build, "
                            "all docs will be rebuilt"
                        ),
                        self.templates.newest_template_name(),
```